backend/src/appointment/database/models.py

```diff
@@ -96,9 +96,17 @@ class TimeMode(enum.Enum):
     h24 = 24
 
 
+def calculate_encrypted_length(length: int):
+    """Calculate the length of the string after it's been encrypted and encoded."""
+    cipher_length = length + 16 - (length % 16)  # Fixed block size of 16 for Aes
+    return int((cipher_length + 2) / 3) << 2  # Base64 with padding
+
+
 def encrypted_type(column_type, length: int = 255, **kwargs) -> StringEncryptedType:
     """Helper to reduce visual noise when creating model columns"""
-    return StringEncryptedType(column_type, secret, AesEngine, 'pkcs5', length=length, **kwargs)
+    return StringEncryptedType(
+        column_type, secret, AesEngine, 'pkcs5', length=calculate_encrypted_length(length), **kwargs
+    )
```

Comment:
Somehow I always thought, that the StringEncryptedType would take care of the appropriate length automatically, but well, that was a little too naive it seems 😅

Reply:
same 😅